File: LV 2/다음 큰 숫자.py
def getonecount(n):
    result = 0
    while(n > 0):
        if n%2 == 1:
            result += 1
        n = n // 2
    return result

def solution(n):
    answer = 0
    oc = getonecount(n)
    for i in range(n+1, 1000000):
        if oc == getonecount(i):
            answer = i
            break
    
    return answer

File: LV 2/다음 큰 숫자.java
class Solution {

    public String toBinary(int n) {
        return Integer.toBinaryString(n);
    }

    public int getOneCount(String s) {
        int count = 0;
        for (char c : s.toCharArray()) {
            if (c == '1') {
                count += 1;
            }
        }
        return count;
    }

    public int solution(int n) {
        int originalCount = getOneCount(toBinary(n));
        int current = n+1;
        while (true) {
            int oneCount = getOneCount(toBinary(current));
            if (oneCount == originalCount) {
                break;
            }
            current += 1;
        }

        return current;
    }
}
